### ISSUE TITLE
[Quest] A Necessary Distraction (BB #1002)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#1002]
This issue was migrated from bitbucket.
**Original Reporter:** soniuch
**Original Date:** 11.06.2015 14:30:13 GMT+0000
**Original Priority:** major
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1002

<hr>

used the item http://www.wowhead.com/item=30811 on Azaloth not releasing him.

http://www.wowhead.com/quest=10688

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 13.06.2015 13:47:29 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1002/#comment-18789405

<hr>

```
#!c++

/*######
## npc_azaloth
######*/

struct npc_azalothAI : public ScriptedAI
{
    bool free;
    uint64 PlayerGUID;

    void Reset()
    {
      free = false;
      PlayerGUID = 0;
      me->CombatStop();
      me->DeleteThreatList();
      me->SetFlag(UNIT_FIELD_FLAGS,UNIT_FLAG_PASSIVE);
      me->SetFlag(UNIT_FIELD_FLAGS,UNIT_FLAG_STUNNED);
      me->SetFlag(UNIT_FIELD_FLAGS,UNIT_FLAG_PACIFIED);
      me->setFaction(1813);
    }

    npc_azalothAI(Creature* c) : ScriptedAI(c){}

     void SpellHit(Unit* caster, const SpellEntry* spell)
    {
        if (caster->GetTypeId() == TYPEID_PLAYER && spell->Id == 37834 && !free)
        {
            free = true;
            PlayerGUID = caster->GetGUID();
            me->RemoveFlag(UNIT_FIELD_FLAGS,UNIT_FLAG_PASSIVE);
            me->RemoveFlag(UNIT_FIELD_FLAGS,UNIT_FLAG_STUNNED);
            me->RemoveFlag(UNIT_FIELD_FLAGS,UNIT_FLAG_PACIFIED);
            me->RemoveAllAuras();
            me->setFaction(954);

            Unit* Warlock = me->FindNearestCreature(21503, 50);

            if (Warlock)
            {
                me->AddThreat(Warlock, 100000.0f);
                AttackStart(Warlock);
            }
            DoMeleeAttackIfReady();
        }
    }

    void UpdateAI(const uint32 diff)
    {
      Unit* Warlock = me->FindNearestCreature(21503, 50);

      if (!Warlock && !UpdateVictim())
      {
        me->ForcedDespawn();
      }

      if (me->isInCombat() && me->getVictim() && me->HasAura(37833,0))
      {
        if (me->getVictim()->GetTypeId() == TYPEID_PLAYER)
        {
          me->StopMoving();
          me->AttackStop();
          Unit *victim = me->getVictim();
          me->CombatStop();
          me->DeleteThreatList();
          return;
        }
      }
      DoMeleeAttackIfReady();
    }
};

CreatureAI* GetAI_npc_azaloth(Creature* pCreature)
{
    return new npc_azalothAI(pCreature);
}

/*######
## npc_sunfurywarlock
######*/

enum sunfurywarlock
{
  SPELL_BEAM_BANISH = 39123,
  NPC_AZALOTH = 21506,
}; 

struct npc_sunfurywarlockAI : public ScriptedAI
{
  bool CastBeam;

  void Reset()
  {
    CastBeam = false;
  }

  npc_sunfurywarlockAI(Creature* c) : ScriptedAI(c) {}

  void EnterCombat(Unit* who)
  {
    me->InterruptNonMeleeSpells(true);
  }

  void UpdateAI(const uint32 diff)
  {
    if (Unit* azaloth = me->FindNearestCreature(NPC_AZALOTH, 50.0f))
    {
      if (azaloth->isAlive() && !CastBeam && !me->isInCombat() && azaloth->HasAura(37833,0))
        {
          CastBeam = true;
          DoCast(azaloth, SPELL_BEAM_BANISH);
        }
    }
    DoMeleeAttackIfReady();

    if (!UpdateVictim())
      return;
  }

  void MoveInLineOfSight(Unit* who)
  {
    if (Unit* pCombat = me->FindNearestCreature(21506, 50.0f))
    {
      if (!pCombat->HasAura(37833, 0))
        AttackStart(pCombat);
    } 
  }
};

CreatureAI* GetAI_npc_sunfurywarlock(Creature* pCreature)
{
  return new npc_sunfurywarlockAI (pCreature);
}

    newscript = new Script;
    newscript->Name = "npc_azaloth";
    newscript->GetAI = &GetAI_npc_azaloth;
    newscript->RegisterSelf();

    newscript = new Script;
    newscript->Name = "npc_sunfurywarlock";
    newscript->GetAI = &GetAI_npc_sunfurywarlock;
    newscript->RegisterSelf();
```

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 17.06.2015 12:49:54 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1002/#comment-18879474

<hr>

ok?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 21.07.2015 14:07:25 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1002/#comment-20361187

<hr>

[Scripts/Quests] Add script support for quest A Necessary Distraction.

```
             Fixes #1002
             Thanks to PadreWoW for the original.
```

→ <<cset d436a965334c>>
